### PR TITLE
Refactor MapLayers UI

### PR DESCRIPTION
**Pull request by aruniverse**
- remove unecessary WidgetControl
- have UiProviders register ui for both (ui-1.0 & 2.0)

### CODE REVIEW

File: test-apps/ui-test-app/src/frontend/index.tsx
Comment (bsteinbk) [inline test-apps/ui-test-app/src/frontend/index.tsx:401]: With latest PR #3242 the try/catch block should be replaced with 
```
const schemaLocater = new ECSchemaRpcLocater(iModelConnection);
await IModelApp.quantityFormatter.setUnitsProvider (new SchemaUnitProvider(schemaLocater));
```

**Review by mdastous-bentley — APPROVED** [review]

File: extensions/map-layers/src/ui/MapFeatureInfoTool.ts
Comment (aruniverse) [inline extensions/map-layers/src/ui/MapFeatureInfoTool.ts:54]: @mdastous-bentley , does it make more sense for users to define their own event and set MapLayersUi.onMapHit to that, or for us to define it and for them to just call `MapLayersUi.onMapHit.raiseEvent`? This is the only coupling left that im iffy about

Comment (mdastous-bentley) [reply extensions/map-layers/src/ui/MapFeatureInfoTool.ts:54]: @aruniverse 
Yea I thought of that that,  I agree there is something weird about the API.  We can certainly 'own' the event, but I thought it would be confusing: generally when an object expose an event, it allows consumers to subscribe to this event and not fire it.  

So maybe we should keep this event internal, and expose a function/class with a method like `fetchFeatureInfoData(mapHit:HitDetails)`

Would it make more sense?

**Comment by aruniverse** [comment]
Publishing as I'd like these changes in 3.1. This PR removes a new widget control that was added to master since 3.0. Widget controls are a deprecated ui-1.0 concept, and I don't want to be publishing any new "deprecated" features/components.

File: docs/changehistory/NextVersion.md
Comment (pmconne) [inline docs/changehistory/NextVersion.md:35]: Convert "MapLayersUI" into a docs link; note that it's a `@beta` API since you're breaking it.

**Review by pmconne — APPROVED** [review]
Probably want to hold off on #3256 until this one's merged.

File: docs/changehistory/NextVersion.md
Comment (pmconne) [inline docs/changehistory/NextVersion.md:46]: Neither of your two snippets of sample code look like they would compile. Please use working code illustrating how existing code can be upgraded to the new API.

File: docs/changehistory/NextVersion.md
Comment (pmconne) [reply docs/changehistory/NextVersion.md:35]: Actually...this is an Extension. It doesn't have an API, does it?

Comment (aruniverse) [reply docs/changehistory/NextVersion.md:35]: tbh im not sure how i feel about this being called an "Extension". Eventually we'd want this to be built with the extension api, but atm it doesnt and its misleading. 

I didnt even realize we never had any docs for map-layers :/

Comment (pmconne) [reply docs/changehistory/NextVersion.md:35]: Yeah, it's weird. ui-test-app consumes it as a package and uses its API. It has an empty docs script, no extract-api script, yet (some) APIs have release tags.
Perhaps it should not be mentioned in NextVersion.md at all until it's been sorted out.

File: extensions/map-layers/src/ui/MapFeatureInfoTool.ts
Comment (aruniverse) [reply extensions/map-layers/src/ui/MapFeatureInfoTool.ts:54]: Mind taking a look at the latest commit where i attempt to decouple the event from the singleton?

File: test-apps/ui-test-app/src/frontend/index.tsx
Comment (bsteinbk) [inline test-apps/ui-test-app/src/frontend/index.tsx:394]: @aruniverse  The removed code (without the try/catch) was based on Chris Lawson's latest change to remove the need for the application to have to reset the units provider.  Did you get exceptions and that is why you added the try/catch back in?

Comment (aruniverse) [reply test-apps/ui-test-app/src/frontend/index.tsx:394]: i think this is left over from a previous merge of mine, ill remove the try/catch

**Review by bsteinbk — APPROVED** [review]

File: extensions/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx
Comment (mdastous-bentley) [inline extensions/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx:18]: We might want to make `_featureInfoOpts ` mandatory since onMapHit should be mandatory.

File: extensions/map-layers/src/ui/Interfaces.ts
Comment (mdastous-bentley) [inline extensions/map-layers/src/ui/Interfaces.ts:53]: I think this should be mandatory, FeatureInfoWidget wont work without this event.  By making it optional, you might confuse developers.

File: extensions/map-layers/src/ui/MapFeatureInfoTool.ts
Comment (mdastous-bentley) [reply extensions/map-layers/src/ui/MapFeatureInfoTool.ts:54]: @aruniverse added comments.

File: extensions/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx
Comment (aruniverse) [reply extensions/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx]: updated

File: extensions/map-layers/src/ui/Interfaces.ts
Comment (aruniverse) [reply extensions/map-layers/src/ui/Interfaces.ts]: updated

File: extensions/map-layers/src/ui/MapFeatureInfoTool.ts
Comment (aruniverse) [reply extensions/map-layers/src/ui/MapFeatureInfoTool.ts:54]: discussed/tested and works fine; resolving comment

File: docs/changehistory/NextVersion.md
Comment (aruniverse) [reply docs/changehistory/NextVersion.md]: Thanks for calling this out @pmconne ; I made sure to add an AHA item for "extension" documentation to tackle in the coming weeks. Reverted my changes here until we have a solution.

**Pull request merged**
